airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -353,70 +354,98 @@ def read_records(
             raise Exception(f"Unknown response for stream `{self.name}`. Response body {report_payload}")
 
 
-class MerchantListingsReports(ReportsAmazonSPStream):
+class IncrementalReportsAmazonSPStream(ReportsAmazonSPStream):
+    @property
+    @abstractmethod
+    def cursor_field(self) -> Union[str, List[str]]:
+        pass
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_benchmark = latest_record[self.cursor_field]
+        if current_stream_state.get(self.cursor_field):
+            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
+        return {self.cursor_field: latest_benchmark}
+
+
+class MerchantListingsReports(IncrementalReportsAmazonSPStream):
     name = "GET_MERCHANT_LISTINGS_ALL_DATA"
+    primary_key = "listing-id"
+    cursor_field = "dataEndTime"
 
 
-class NetPureProductMarginReport(ReportsAmazonSPStream):
+class NetPureProductMarginReport(IncrementalReportsAmazonSPStream):
     name = "GET_VENDOR_NET_PURE_PRODUCT_MARGIN_REPORT"
+    cursor_field = "dataEndTime"
 
 
-class RapidRetailAnalyticsInventoryReport(ReportsAmazonSPStream):
+class RapidRetailAnalyticsInventoryReport(IncrementalReportsAmazonSPStream):
     name = "GET_VENDOR_REAL_TIME_INVENTORY_REPORT"
+    cursor_field = "dataEndTime"
 
 
-class FlatFileOrdersReports(ReportsAmazonSPStream):
+class FlatFileOrdersReports(IncrementalReportsAmazonSPStream):
     """
     Field definitions: https://sellercentral.amazon.com/gp/help/help.html?itemID=201648780
     """
 
     name = "GET_FLAT_FILE_ALL_ORDERS_DATA_BY_ORDER_DATE_GENERAL"
+    cursor_field = "dataEndTime"
 
 
-class FbaStorageFeesReports(ReportsAmazonSPStream):
+class FbaStorageFeesReports(IncrementalReportsAmazonSPStream):
     """
     Field definitions: https://sellercentral.amazon.com/help/hub/reference/G202086720
     """
 
     name = "GET_FBA_STORAGE_FEE_CHARGES_DATA"
+    cursor_field = "dataEndTime"
 
 
-class FulfilledShipmentsReports(ReportsAmazonSPStream):
+class FulfilledShipmentsReports(IncrementalReportsAmazonSPStream):
     """
     Field definitions: https://sellercentral.amazon.com/gp/help/help.html?itemID=200453120
     """
 
     name = "GET_AMAZON_FULFILLED_SHIPMENTS_DATA_GENERAL"
+    cursor_field = "dataEndTime"
```

Comment:
Let's move the `cursor_field` to the base class and override only when it needs.